fix: small leftover issues with new starter screen

urmauur opened the pull request:
## Describe Your Changes

1. Disable the "New Threads" icon on the top
![image](https://github.com/user-attachments/assets/ea747944-3bac-43b2-a359-6f42e8d11a5d)

2. Duplicate render progress bar when more than one download from feat/suggestion model.
<img width="1032" alt="Screenshot 2024-09-14 at 23 01 20" src="https://github.com/user-attachments/assets/03bf1a58-0092-4fdb-94db-a91f356539bc">

3. @louis-jan temporary with hard-coded recommend model

4. Overlap progress bar when dropdown search open

![image](https://github.com/user-attachments/assets/6e840a29-f00f-4045-aaad-bec1a4c53e69)

5. Cloud model should be left aligned
![image](https://github.com/user-attachments/assets/e9b754e8-8c0d-4938-a3ee-bc0b265d84e7)

6. Update UI regarding miss font weight and spacing
![image](https://github.com/user-attachments/assets/aabf35f6-03af-4ca4-8d35-bc7fc747c479)



### Changes code:

1. Update conditional render `New threads` icon when user downloads a local model or configured any cloud model and thread length 
<img width="1038" alt="Screenshot 2024-09-15 at 00 34 40" src="https://github.com/user-attachments/assets/257b9030-d0b3-464d-b9b2-7ee005e11a93">

2. The new code filters the `downloadStates` object values using the condition `x.modelId === featModel.id`. This means that only items with a `modelId` matching `featModel.id` will be rendered.
<img width="1042" alt="Screenshot 2024-09-14 at 23 24 14" src="https://github.com/user-attachments/assets/91508ffe-0471-4b84-b20a-e23115155911">

3. temporary with hard-coded recommend model 

4. Update `z-index` to avoid overlap, since the element dropdown options is absolute 
<img width="1055" alt="Screenshot 2024-09-15 at 00 42 34" src="https://github.com/user-attachments/assets/a5f18c43-aca4-46a0-a600-5cbc14aedddf">

5. Removed justify center flex 
 <img width="1119" alt="Screenshot 2024-09-15 at 00 47 24" src="https://github.com/user-attachments/assets/dc81aeeb-ff74-47f8-a410-9037199d5a55">

6. Change some class font weight and spacing
<img width="1094" alt="Screenshot 2024-09-15 at 01 01 35" src="https://github.com/user-attachments/assets/f5b94fda-5db4-440a-a226-5f65a3ce7a85">


## Fixes Issues

- Closes #3660  
- Closes #

## Self Checklist

- [ ] Added relevant comments, esp in complex areas
- [ ] Updated docs (for bug fixes / features)
- [ ] Created issues for follow-up changes or refactoring needed

dan-homebrew reviewed (approved):
Approving to unblock, and will let @imtuyethan @eckartal find issues during QA process

dan-homebrew reviewed (approved):
Approving to unblock. Will let @imtuyethan @eckartal find issues during Manual QA

dan-homebrew commented:
@urmauur Can you look into why the Playwright tests are failing?

urmauur commented:
> @urmauur Can you look into why the Playwright tests are failing?

Hi @dan-homebrew and @louis-jan,
Can you check again this PR?

The issue with the PR failure is in the test `Select GPT model from Hub and Chat with Invalid API Key`. Since we updated the thread with the starter screen, the “Create Thread” button is no longer rendered, so the `btn-create-thread` selector is missing.

However, the `btn-create-thread` selector is not necessary because the test scenario follows this flow:

- Go to the hub page
- Select cloud mode
- Send a chat
- Expect the `APIKeyError` component to render
	
removed this code 

```tsx
await page
    .getByTestId('btn-create-thread')
    .click()
```

urmauur commented:
> @urmauur Can you look into why the Playwright tests are failing?

Hi @dan-homebrew and @louis-jan,
Can you check again this PR?

The issue with the PR failure is in the test `Select GPT model from Hub and Chat with Invalid API Key`. Since we updated the thread with the starter screen, the “Create Thread” button is no longer rendered, so the `btn-create-thread` selector is missing.

However, the `btn-create-thread` selector is not necessary because the test scenario follows this flow:

- Go to the hub page
- Select cloud mode
- Send a chat
- Expect the `APIKeyError` component to render
	
removed this code 

```tsx
await page
    .getByTestId('btn-create-thread')
    .click()
```

urmauur commented on web/screens/Thread/index.test.tsx:
```diff
@@ -0,0 +1,35 @@
+import React from 'react'
```
Add a unit test here to ensure the thread screen renders correctly

urmauur commented on electron/tests/e2e/thread.e2e.spec.ts:
```diff
@@ -1,32 +1,29 @@
   // Select the first GPT model
   await page
     .locator('[data-testid^="use-model-btn"][data-testid*="gpt"]')
-    .first().click()
```
Removed unnecessary selector

urmauur commented on electron/tests/e2e/thread.e2e.spec.ts:
```diff
@@ -1,32 +1,29 @@
-
-  // Attempt to create thread and chat in Thread page
-  await page
-    .getByTestId('btn-create-thread')
```
Removed unnecessary selector

louis-jan reviewed (approved):
LGTM

Just need to find somehow to define recommended models better.
